biz.aQute.junit.platform/src/aQute/junit/platform/Activator.java

```diff
@@ -0,0 +1,390 @@
+			thread.setContextClassLoader(l);
+		}
+
+		List<TestExecutionListener> listenerList = new ArrayList<>();
+
+		String testcases = context.getProperty(TESTER_NAMES);
+		trace("test cases %s", testcases);
+		int port = -1;
+		boolean rerunIDE = false;
+		if (context.getProperty(TESTER_CONTROLPORT) != null) {
+			port = Integer.parseInt(context.getProperty(TESTER_CONTROLPORT));
+			rerunIDE = true;
+		} else if (context.getProperty(TESTER_PORT) != null) {
+			port = Integer.parseInt(context.getProperty(TESTER_PORT));
+		}
+
+		if (port > 0) {
+			try {
+				trace("using control port %s, rerun IDE?: %s", port, rerunIDE);
+				jUnitEclipseReport = new JUnitEclipseListener(port, rerunIDE);
+				listenerList.add(jUnitEclipseReport);
+			} catch (Exception e) {
+				System.err.println(
+					"Cannot create link Eclipse JUnit control on port " + port + " (rerunIDE: " + rerunIDE + ')');
+				System.exit(254);
+			}
+		}
+
+		String testerDir = context.getProperty(TESTER_DIR);
+		if (testerDir == null)
+			testerDir = "testdir";
+
+		reportDir = new File(testerDir);
+
+		//
+		// Jenkins does not detect test failures unless reported
+		// by JUnit XML output. If we have an unresolved failure
+		// we timeout. The following will test if there are any
+		// unresolveds and report this as a JUnit failure. It can
+		// be disabled with -testunresolved=false
+		//
+		unresolved = context.getProperty(TESTER_UNRESOLVED);
+
+		trace("run unresolved %s", unresolved);
+
+		if (!reportDir.exists() && !reportDir.mkdirs()) {
+			System.err.printf("Could not create directory %s%n", reportDir);
+		} else {
+			trace("using %s, path: %s", reportDir, reportDir.toPath());
+			try {
+				listenerList.add(new JUnitXmlListener(reportDir.toPath(), new PrintWriter(System.err)));
+			} catch (Exception e) {
+				error("Error trying to create xml reporter: %s", e);
+			}
+		}
+
+		basicReport = LoggingListener.forBiConsumer((t, msg) -> {
+			if (t == null) {
+				trace(msg.get());
+			} else {
+				trace(msg.get(), t);
+			}
+		});
+		summary = new SummaryGeneratingListener();
+		listenerList.add(basicReport);
+		listenerList.add(summary);
+		listeners = listenerList.stream()
+			.toArray(TestExecutionListener[]::new);
+
+		if (testcases == null) {
+			trace("automatic testing of all bundles with " + aQute.bnd.osgi.Constants.TESTCASES + " header");
+			try {
+				automatic();
+			} catch (IOException e) {
+				// ignore
+			}
+		} else {
+			trace("receivednames of classes to test %s", testcases);
+			try {
+				baseSelectors = Stream.of(testcases.split("\\s*,\\s*"))
+					.map(x -> {
+						if (x.contains(":")) {
+							return selectMethod(x.replace(':', '#'));
+						} else {
+							return selectClass(x);
+						}
+					})
+					.collect(Collectors.toList());
+				automatic();
+			} catch (Exception e) {
+				e.printStackTrace();
+				System.exit(254);
+			}
+		}
+	}
+
+	void automatic() throws IOException {
+		final List<LauncherDiscoveryRequest> queue = new Vector<>();
```

Comment:
This use of Vector was just replaced with BlockingDeque. https://github.com/bndtools/bnd/commit/147ee99eec3006006f5023100c84668303c8ad9b

Reply:
Makes sense, fixed.